code/src/components/Header.js

```diff
@@ -0,0 +1,94 @@
+import React from 'react'
```

Comment:
This comment refers to whole file.
I want to congratulate for building very complex logic, you did great job :)
Nonetheless, I want to be clear about one thing - this is not how we build this type of logic in React. If we would have to do this "by book", we would have used React state. But since it was not yet introduced to you, I completely understand different approach you took over here.
I think that in 1-2 weeks from now on, you will fully see how easy (and far more short!) this logic could be when written with React state :)